src/main/java/seedu/address/model/student/Id.java

```diff
@@ -1,31 +1,63 @@
 package seedu.address.model.student;
 
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.Set;
+
 /**
  * Represents a Student's id.
  */
 public class Id {
-    public static final String MESSAGE_CONSTRAINTS = "Id must be numeric.";
+    public static final String MESSAGE_CONSTRAINTS = "Id must be numeric and unique.";
 
     /** pads the Id value with leading spaces to be converted to zeroes */
     public static final String PADDING_FORMAT = "%1$3s";
 
     /** validates if the string is numeric */
     public static final String VALIDATION_REGEX = "-?\\d+(\\.\\d+)?";
 
+    /** fields allowing auto-assignment and uniqueness of Id value */
+    private static int lastUsedId = 0;
+    private static final Set<String> usedIds = new HashSet<>();
+
     public final String value;
```

Comment:
shouldn't this be private not public?

Reply:
it's public like the other student field classes